vars/parseStageInfo.groovy

```diff
@@ -214,8 +214,19 @@ void call(Map config = [:]) {
         result['pragma_suffix'] = '-hw-small'
       } else if (stage_name.contains('Medium')) {
         result['node_count'] = 5
-        cluster_size = 'hw,medium'
+        cluster_size = 'hw,medium,-provider'
         result['pragma_suffix'] = '-hw-medium'
+        if (stage_name.contains('Provider')) {
+          cluster_size = 'hw,medium,provider'
+          if (stage_name.contains('Verbs')) {
+            result['pragma_suffix'] += '-verbs-provider'
+            ftest_arg_provider = 'ofi+verbs'
+          }
+          elif (stage_name.contains('UCX')) {
+            result['pragma_suffix'] += '-ucx-provider'
+            ftest_arg_provider = 'ucx+dc_x'
+          }
+        }
```

Comment:
A few points
1. Is the syntax `elif` or `else if`? I see `else if` above in this file.
2. Do we want an `else` just in case?
3. Just a thought - I've always thought the Skip-* pragmas would be better served if we used some sort of "subset" naming convention. I.e.
    - `hw-medium`
    - `hw-medium-provider-verbs`
    - `hw-medium-provider-ucx`
    And skip pragmas:
    - `Skip-test-func-hw-medium` - Skips all `hw-medium*`
    - `Skip-test-func-hw-medium-provider` - Skips all `hw-medium-provider*`
    - `Skip-test-func-hw-medium-provider-ucx` - Skips all `hw-medium-provider-ucx*`
    But for this to really shine, we'd want to refactor the existing pragmas. A benefit is the logic for skipping groups of stages becomes MUCH simpler. Not asking you to do this - just something to think about.

Reply:
> A few points
> 
> 1. Is the syntax `elif` or `else if`? I see `else if` above in this file.

Good  catch @daltonbohning.  The syntax is indeed `else if`.

Reply:
Fixed, thanks guys.